jormungandr/src/fragment/pool.rs

```diff
@@ -305,6 +307,22 @@ impl Pools {
             block_date.into(),
         );
     }
+
+    async fn get_current_block_date(&self) -> BlockDate {
+        let time = std::time::SystemTime::now();
+        let tip = self.tip.get_ref().await;
+        let era = tip.epoch_leadership_schedule().era();
+        let epoch_position = tip
+            .time_frame()
+            .slot_at(&time)
+            .and_then(|slot| era.from_slot_to_era(slot))
+            .expect("the current time and blockchain state should produce a valid blockchain date");
+        let block_date: BlockDate = epoch_position.into();
+        BlockDate {
+            slot_id: block_date.slot_id + 1,
```

Comment:
I think it could happen that a transaction is received just at the slot start before block creation starts and is incorrectly mislabeled as expired when in theory it could still make its way into the block. In any case, I think this is unlikely and anyway hard to distinguish from practical limits in clocks synchronization among parties